lib/sinatra/indifferent_hash.rb

```diff
@@ -89,34 +89,41 @@ def key?(key)
       super(convert_key(key))
     end
 
-    alias_method :has_key?, :key?
-    alias_method :include?, :key?
-    alias_method :member?, :key?
+    alias has_key? key?
+    alias include? key?
+    alias member? key?
 
     def value?(value)
       super(convert_value(value))
     end
 
-    alias_method :has_value?, :value?
+    alias has_value? value?
 
     def delete(key)
       super(convert_key(key))
     end
 
+    # Added in Ruby 2.3
     def dig(key, *other_keys)
       super(convert_key(key), *other_keys)
     end
 
-    def fetch_values(*keys)
-      keys.map!(&method(:convert_key))
+    # Added in Ruby 2.3
+    if method_defined?(:fetch_values)
+      def fetch_values(*keys)
+        keys.map!(&method(:convert_key))
 
-      super(*keys)
+        super(*keys)
+      end
     end
 
-    def slice(*keys)
-      keys.map!(&method(:convert_key))
+    # Added in Ruby 2.5
+    if method_defined?(:slice)
```

Comment:
This `method_defined?` check isn't present in master, it was removed because we support only ruby 2.6 onwards, same for `fetch_values`

Reply:
Good catch 👍